packages/next/src/client/components/router-reducer/fetch-server-response.ts

```diff
@@ -84,6 +84,7 @@ export async function fetchServerResponse(
       headers[NEXT_ROUTER_PREFETCH_HEADER] || '0',
       headers[NEXT_ROUTER_STATE_TREE],
       headers[NEXT_URL],
+      currentBuildId,
```

Comment:
I have a [branch](https://github.com/vercel/next.js/commit/cb298787dabf7396d908a790ce2af0756f108401) which also adds the build id to the Vary header but it's hard to reconcile that with Vercel's promises for [skew protection](https://vercel.com/docs/deployments/skew-protection) or the `?dpl` (which doesn't seem to be appended for RSC payload requests, even with `deploymentId` set in the config). This route seemed overly complicated and unnecessary.

The value here is still just a unique search query as described when the param was added in https://github.com/vercel/next.js/pull/50970 ... just more unique now.